tests/e2e/utils/PreferencesHandler.ts

```diff
@@ -63,7 +63,7 @@ export class PreferencesHandler {
         await this.setPreference('vs-kubernetes', vsKubernetesConfig);
     }
 
-    private async setPreference(attribute: string, value: any) {
+    public async setPreference(attribute: string, value: any) {
```

Comment:
why not create a public `setShellcheckPathPreference` method?
I don't like this change from private to public on a method that does not need to be public.

If you really want to change it to public, then please remove the other methods and change it in other tests and CodereadyReporter as well, so that the approach is uniform everywhere.

Reply:
Could you please clarify your point:
 - What is an advantage to create method in the common page object for one particular test?
 - For what remove other methods? Which probblem it resolves?

Reply:
In short, consistency. In long, consistency.

Everywhere else in the code, we use PreferencesHandler's public method that we implement for a specific purpose, which also allows us to do some extra logic around it. E.g.: you could implement your use case as 
```
public async setShellcheckPathPreference() {
  const shellcheckExecutablePathPropertyName: string = 'shellcheck.executablePath';
  const shellcheckExecutablePath: string = '/bin/shellcheck';

 await setPreference(shellcheckExecutablePathPropertyName, shellcheckExecutablePath);
}
```
and call it in your test as part of `test('Create workspace using factory'`
```
{
  await preferencesHandler.setShellcheckPathPreference();
  await driverHelper.navigateToUrl(factoryUrl);
}
```
which would make it consistent with the rest of the codebase.
setPreference was an internal method of the class that shouldn't be accessed from the outside.

for your second point, you can do search for all the other methods from PreferencesHandler in the codebase and replace them with raw preferenceHandler.setPreference calls if you want to have it as a public method.

I really don't like that we're introducing fragmentation into the code as it makes it harder to debug and maintain.

Reply:
Thank you Tibor, but from your post I can't get answer to my questions:
 - which problem resolves the private "setPreference" method and why not use it directly
 - why I need to create a dedicated method for one particular test, which will not be reused, in the common pageobject. In my eyes it look like not a very right approach.

Reply:
I haven't read whole slack conversation about this... But looking at the PR and arguments written here I personally think `setPreference` method should be public and should be used as much as possible (but current usages of the other methods shouldn't be changed in this PR).
I personally think that havin extra method for each separate preference we might want to configure doesn't scale and thus this approach should be used only in rare circumnstances, where there would be some other logic involved (a.k.a. don't create methods just as a  "named" wrappers for `setPreference`). In most of the cases it's ok to call `serPreference` directly from tests or other places, as the arguments should pretty much cover what preference is being set (thus no need to have separate method for this in here)

Reply:
I'm unblocking this change after our discussion on BlueJeans, Method will be changed to public and we will create a new issue to track the removal of wrapper methods. (Possible issue for Intern)

Reply:
Thank you @ScrewTSW  here is an issue: "Get rid of wrappers in the "PreferencesHandler" class" https://github.com/eclipse/che/issues/19208